### PR TITLE
Added testing for c++17 to latest compilers with test with

### PR DESCRIPTION
**Pull request by annagrin**
Added running latest compilers with -std=c++17 option to CI the test matrix,
Updated cmake configuration to allow passing c++ standard on the command line.

### CODE REVIEW

**Comment by annagrin** [comment]
this resolves #694

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,12 @@ else()
     add_custom_target(catch)
 endif()
 
+if (MSVC)
+    if (GSL_CXX_STANDARD EQUAL 17)
```
Comment (johelegp) [inline tests/CMakeLists.txt]: This can be merged with the above to become `MSVC AND GSL_CXX_STANDARD EQUAL 17`.

Comment (annagrin) [reply tests/CMakeLists.txt]: yes thanks

**Comment by johelegp** [comment]
Look at this error from https://ci.appveyor.com/project/neilmacintosh/gsl/build/0.1.812/job/8cl8bcojnq48np37.
>c:\projects\gsl\tests\utils_tests.cpp(121): error C2429: language feature 'terse static assert' requires compiler flag '/std:c++17' [C:\projects\gsl\build\tests\utils_tests.vcxproj]

Is it really compiling with C++17 when `GSL_CXX_STANDARD` is set to `17`?

**Pull request merged**